### PR TITLE
NIFI-11977 Set UTF-8 Character Set in Subject for PutEmail

### PR DESCRIPTION
**Pull request by makinyemi**
# Summary

[NIFI-11977](https://issues.apache.org/jira/browse/NIFI-11977)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java [hunk 1]
```diff
@@ -431,7 +431,12 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 }
             }
             this.setMessageHeader("X-Mailer", context.getProperty(HEADER_XMAILER).evaluateAttributeExpressions(flowFile).getValue(), message);
-            message.setSubject(context.getProperty(SUBJECT).evaluateAttributeExpressions(flowFile).getValue());
+
+            if (message instanceof MimeMessage) {
+                ((MimeMessage) message).setSubject(context.getProperty(SUBJECT).evaluateAttributeExpressions(flowFile).getValue(), StandardCharsets.UTF_8.name());
+            } else {
+                message.setSubject(context.getProperty(SUBJECT).evaluateAttributeExpressions(flowFile).getValue());
+            }
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java:439]: This `instanceof` check is not necessary, the variable should be declared as `MimeMessage` and that should simplify the approach.

```suggestion
message).setSubject(context.getProperty(SUBJECT).evaluateAttributeExpressions(flowFile).getValue(), StandardCharsets.UTF_8.name());
```

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the contribution @makinyemi. As noted in the detailed comment, the `instanceof` check can be removed if the initial variable is declared as `MimeMessage`.

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java (hunk 1 above)
Comment (makinyemi) [reply nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java:439]: Understood. Will make the change

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the update, but I was referring to where the `message` variable is declared earlier in the method. That is declared as `Message` right now, but it should be changed to `MimeMessage`, making the cast unnecessary.

**Comment by makinyemi** [comment]
My apologies. Jumped the gun there. I will make the correct update

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @makinyemi, the latest version looks good! +1 merging

**Pull request merged**